src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -22,19 +23,25 @@ internal class FunctionDispatcher : IFunctionDispatcher
     {
         private readonly IMetricsLogger _metricsLogger;
         private readonly ILogger _logger;
+        private readonly IEnvironment _environment;
+        private readonly IDisposable _rpcChannelReadySubscriptions;
+        private readonly int _debounceSeconds = 10;
         private IScriptEventManager _eventManager;
         private IEnumerable<WorkerConfig> _workerConfigs;
-        private CreateChannel _channelFactory;
         private ILanguageWorkerChannelManager _languageWorkerChannelManager;
-        private ConcurrentDictionary<string, LanguageWorkerState> _workerStates = new ConcurrentDictionary<string, LanguageWorkerState>();
+        private LanguageWorkerState _workerState = new LanguageWorkerState();
```

Comment:
LanguageWorkerState and the ChannelManager look similar -- they both seem to maintain a collection of LanguageWorkerChannels. Do we need to maintain that in both places?

Reply:
I have planned follow up refactoring item to  split LanguageWorkerState  and Channels. I need to add JobhostLanguageWorkerChannelManager similar to LanguageWorkerChannelManager - which manages lifetime of WebHost level language workers